src/createQuery.ts

```diff
@@ -0,0 +1,69 @@
+import { computed, onScopeDispose, reactive, toRefs, toValue, watch } from "vue";
+import { createManager } from "./createManager";
+import { CreateQueryOptions, DisposableQuery, Query, QueryAction, QueryActionArgs, QueryOptions } from "./types";
+import isEqual from 'lodash.isequal'
+import { isDefined } from "./utilities";
+
+export type QueryFunction = <TAction extends QueryAction>(action: TAction, args: Parameters<TAction>, options?: QueryOptions) => DisposableQuery<TAction>
+export type QueryComposition = <TAction extends QueryAction>(action: TAction, args: QueryActionArgs<TAction>, options?: QueryOptions) => Query<TAction>
+
+export type CreateQuery = {
+  query: QueryFunction,
+  useQuery: QueryComposition
+}
+
+const noop = () => undefined
+
+export function createQuery(options?: CreateQueryOptions): CreateQuery {
+  const manager = createManager(options)
+
+  const query: QueryFunction = (action, args, options) => {
+    const query = manager.subscribe(action, args, options)
+
+    return Object.assign(query, {
+      [Symbol.dispose]: () => {
+        query.unsubscribe()
+      },
```

Comment:
I think this is all we need for dispose to work. But was getting build errors when testing. Might be too new for vite/vitest.

Reply:
I was able to get this working in https://github.com/kitbagjs/query/pull/1/commits/91cfc4caa102a790ac200db1d9a968f384f88163